src/basic/NESTED_INIT-Hip.cpp

```diff
@@ -71,17 +96,18 @@ void NESTED_INIT::runHipVariant(VariantID vid)
     startTimer();
     for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
 
-      auto nested_init_lambda = [=] __device__ (Index_type i, Index_type j, Index_type k) {
-        NESTED_INIT_BODY;
-      };
-
-      dim3 nthreads_per_block(ni, 1, 1);
-      dim3 nblocks(1, nj, nk);
+      dim3 nthreads_per_block(i_block_sz, j_block_sz, k_block_sz);
+      dim3 nblocks(static_cast<size_t>(RAJA_DIVIDE_CEILING_INT(ni, i_block_sz)),
+                   static_cast<size_t>(RAJA_DIVIDE_CEILING_INT(nj, j_block_sz)),
+                   static_cast<size_t>(RAJA_DIVIDE_CEILING_INT(nk, k_block_sz)));
 
-      auto kernel = lambda_hip_kernel<RAJA::hip_thread_x_direct, RAJA::hip_block_y_direct, RAJA::hip_block_z_direct, decltype(nested_init_lambda)>;
-      hipLaunchKernelGGL(kernel,
-        nblocks, nthreads_per_block, 0, 0,
-        0, ni, 0, nj, 0, nk, nested_init_lambda);
+      hipLaunchKernelGGL((nested_init_lam), 
```

Comment:
This won't compile, `nested_init_lam` is just a function template, you have to give it `nested_init_lam<decltype(lambda)>` to get a function pointer.

Reply:
Ah. Thanks. I hadn't gotten to the building and testing the Hip stuff yet. I was wondering about the seemingly overly complex code for Hip vs. CUDA.

Reply:
Another option is to use `(cuda|hip)LaunchKernel` for everything which would be consistent and have the least overhead.

Reply:
I agree we should do that. But in a separate PR.